compiler/aarch64/codegen/ARM64SystemLinkage.cpp

```diff
@@ -516,17 +534,313 @@ int32_t TR::ARM64SystemLinkage::buildArgs(TR::Node *callNode,
+         }
+      }
+
+   // From here, down, any new stack allocations will expire / die when the function returns
+   TR::StackMemoryRegion stackMemoryRegion(*trMemory());
+   /* End result of Step 1 - determined number of memory arguments! */
+   if (numMemArgs > 0)
+      {
+      pushToMemory = new (trStackMemory()) TR::ARM64MemoryArgument[numMemArgs];
+
+      argMemReg = cg()->allocateRegister();
+      }
+
+   totalSize = numMemArgs * 8;
+   // align to 16-byte boundary
+   totalSize = (totalSize + 15) & (~15);
+
+   numIntegerArgs = 0;
+   numFloatArgs = 0;
+
+   for (i = firstArgumentChild; i < callNode->getNumChildren(); i++)
+      {
+      TR::MemoryReference *mref = NULL;
+      TR::Register *argRegister;
+      TR::InstOpCode::Mnemonic op;
+      int32_t mrefSize;
+
+      child = callNode->getChild(i);
+      childType = child->getDataType();
+
+      switch (childType)
+         {
+         case TR::Int8:
+         case TR::Int16:
+         case TR::Int32:
+         case TR::Int64:
+         case TR::Address: // have to do something for GC maps here
+            if (childType == TR::Address)
+               argRegister = pushAddressArg(child);
+            else if (childType == TR::Int64)
+               argRegister = pushLongArg(child);
+            else
+               argRegister = pushIntegerWordArg(child);
+
+            if (numIntegerArgs < properties.getNumIntArgRegs())
+               {
+               if (!cg()->canClobberNodesRegister(child, 0))
+                  {
+                  if (argRegister->containsCollectedReference())
+                     tempReg = cg()->allocateCollectedReferenceRegister();
+                  else
+                     tempReg = cg()->allocateRegister();
+                  generateMovInstruction(cg(), callNode, tempReg, argRegister);
+                  argRegister = tempReg;
+                  }
+               if (numIntegerArgs == 0 &&
+                  (resType.isAddress() || resType.isInt32() || resType.isInt64()))
+                  {
+                  TR::Register *resultReg;
+                  if (resType.isAddress())
+                     resultReg = cg()->allocateCollectedReferenceRegister();
+                  else
+                     resultReg = cg()->allocateRegister();
+
+                  dependencies->addPreCondition(argRegister, TR::RealRegister::x0);
+                  dependencies->addPostCondition(resultReg, TR::RealRegister::x0);
+                  }
+               else
+                  {
+                  addDependency(dependencies, argRegister, properties.getIntegerArgumentRegister(numIntegerArgs), TR_GPR, cg());
+                  }
+               }
+            else
+               {
+               // numIntegerArgs >= properties.getNumIntArgRegs()
+               if (childType == TR::Address || childType == TR::Int64)
+                  {
+                  op = TR::InstOpCode::strpostx;
+                  mrefSize = 8;
```

Comment:
Why is `mrefSize` tracked?  It's not used anywhere...

Reply:
Originally `mrefSize` was added to `argSize`.  I changed the stack offsets for outgoing arguments, and `argSize` is now always incremented by 8.  Removing the unused variable.